src/k8s/pkg/k8sd/features/calico/cleanup.go

```diff
@@ -0,0 +1,88 @@
+package calico
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/canonical/k8s/pkg/snap"
+)
+
+func CleanupNetwork(ctx context.Context, snap snap.Snap) error {
+	output, err := exec.CommandContext(ctx, "ip", "-j", "link", "show").Output()
+	if err != nil {
+		return fmt.Errorf("failed to list network interfaces: %w", err)
+	}
+
+	// Parse the json output of `ip -j link show` to find the interfaces that were created by Calico.
+	type NetworkInterface struct {
+		Name string `json:"ifname"`
+	}
+	var interfaces []NetworkInterface
+	err = json.Unmarshal(output, &interfaces)
+	if err != nil {
+		return fmt.Errorf("failed to parse network interface JSON: %w", err)
+	}
+
+	// Find the interfaces created by Calico
+	for _, iface := range interfaces {
+		// Check if the interface name matches the regex pattern
+		match, err := regexp.MatchString("^vxlan[-v6]*.calico|cali[a-f0-9]*$", iface.Name)
+		if err != nil {
+			return fmt.Errorf("failed to match regex pattern: %w", err)
+		}
+		if match {
+			// Perform cleanup for Calico interface
+			if _, err := exec.CommandContext(ctx, "ip", "link", "delete", iface.Name).CombinedOutput(); err != nil {
+				return fmt.Errorf("failed to delete interface %s: %w", iface.Name, err)
+			}
+		}
+	}
+
+	// List network namespaces in JSON format
+	nsOutput, err := exec.CommandContext(ctx, "ip", "-j", "netns", "list").Output()
+	if err != nil {
+		return fmt.Errorf("failed to list network namespaces: %w", err)
+	}
```

Comment:
i would greatly recommend listing `/run/netns` or through the `netns` filesystem mounts